### PR TITLE
PR-2

### PR DESCRIPTION
**Pull request by CheezItMan**
This method calculates a factorial

### CODE REVIEW

File: function.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+def function(n)
+  if n == 0
```
Comment (jaitch) [inline function.rb:3]: Did you mean to include "return" before the "1"?

File: function.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+def function(n)
```
Comment (dtaylor73) [inline function.rb:2]: Let's make this function dryer. We could write a ternary statement instead of spreading the code out into multiple lines.

File: function.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  if n == 0
+    1
+  else
+    n * function(n-1)
```
Comment (emilyvomacka) [inline function.rb:6]: Should handle the initial case of n < 0

File: function.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  end
```
Comment (raisahv) [inline function.rb:7]: can shorten this to 

```suggestion
n == 0 ? 1 : n * function(n-1)
```

File: function.rb (hunk 2 above)
Comment (Galaxylaughing) [inline function.rb:2]: `function` doesn't seem like a meaningful name; it should probably be called 'factorial' or something

Comment (north108) [inline function.rb:2]: function naming should be more clear as to what it does

File: function.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+
+def function(n)
+  if n == 0
+    1
```
Comment (dtaylor73) [inline function.rb:4]: Let's write 'puts' in front of the code we want to print to the screen.

File: function.rb (hunk 2 above)
Comment (sun-alice) [inline function.rb:2]: name this something more elaborate

File: function.rb (hunk 5 above)
Comment (Galaxylaughing) [inline function.rb:4]: I think you need to explicitly return 1 here and, on line 6, return `n * function(n-1)`

Comment (north108) [inline function.rb:4]: add return to be more explicit

File: function.rb (hunk 1 above)
Comment (Galaxylaughing) [inline function.rb:3]: if n is a float, this function might go on forever since it will never == zero. Perhaps <= would be better?

File: function.rb (hunk 3 above)
Comment (sun-alice) [inline function.rb:6]:
```suggestion
       return  n * function(n-1)
```

File: function.rb (hunk 2 above)
Comment (dtingg) [inline function.rb:2]: Maybe rename this function to factorial?

File: function.rb (hunk 3 above)
Comment (dtingg) [inline function.rb:6]: If you change the name of the function, you have to change it here too.

File: function.rb (hunk 5 above)
Comment (brilatimer) [inline function.rb:4]: could be clearer

Comment (mdove92) [inline function.rb:4]: What is the 1 supposed to be set to? Are you trying to return it? Please write it out

File: function.rb (hunk 2 above)
Comment (mdove92) [inline function.rb:2]: This should have a more descriptive name. Function dos not let us know what to use it for, and may be a reserved word in ruby.

File: function.rb (hunk 5 above)
Comment (emilyvomacka) [inline function.rb:4]: could add "return" to make it clearer

**Review by Galaxylaughing — COMMENTED** [review]
ok

**Review by raisahv — COMMENTED** [review]
Approved

File: function.rb (hunk 5 above)
Comment (sun-alice) [inline function.rb:4]:
```suggestion
    return 1
```
